src/gwc/src/main/java/org/geoserver/gwc/layer/StyleParameterFilter.java

```diff
@@ -190,20 +194,25 @@ public List<String> getLegalValues() {
     }
 
     /** Set/update the availableStyles and defaultStyle based on the given GeoServer layer. */
-    public void setLayer(LayerInfo layer) {
-        Set<String> newStyles = new TreeSet<>();
-
-        for (StyleInfo style : layer.getStyles()) {
-            newStyles.add(style.prefixedName());
-        }
-
-        availableStyles = newStyles;
-
-        if (layer.getDefaultStyle() != null) {
-            defaultStyle = layer.getDefaultStyle().prefixedName();
+    public void setLayer(PublishedInfo layer) {
+        checkArgument(layer instanceof LayerInfo || layer instanceof LayerGroupInfo);
+
+        StyleInfo defStyle;
+        Collection<StyleInfo> styles;
+        if (layer instanceof LayerInfo) {
+            defStyle = ((LayerInfo) layer).getDefaultStyle();
+            styles = ((LayerInfo) layer).getStyles();
         } else {
-            defaultStyle = null;
+            defStyle = ((LayerGroupInfo) layer).getRootLayerStyle();
```

Comment:
I don't believe this is correct... the "root layer" is a special layer used in WMS for EO (you're picking its style), while the list of styles you're retrieving in the next line is the set of styles associated to each layer in the group.

In both cases, these styles cannot be associated to the group as a whole
Layer group have "styles" which are full alternative definitions of the layer group, either with the same layers but different per layer styles, or with different layers altogether. Each one of these is a "LayerGroupStyle".

What you need to do for layer groups, is to call getLayerGrouStyles and collect their names instead (careful, it could be empty, in that case no alternative style is supported).